arbnode/transaction_streamer.go

```diff
@@ -264,35 +268,33 @@ func deleteFromRange(ctx context.Context, db ethdb.Database, prefix []byte, star
 
 // The insertion mutex must be held. This acquires the reorg mutex.
 // Note: oldMessages will be empty if reorgHook is nil
-func (s *TransactionStreamer) reorg(batch ethdb.Batch, count arbutil.MessageIndex, newMessages []arbostypes.MessageWithMetadataAndBlockHash) error {
-	if count == 0 {
-		return errors.New("cannot reorg out init message")
-	}
-	lastDelayedSeqNum, err := s.getPrevPrevDelayedRead(count)
+func (s *TransactionStreamer) reorg(batch ethdb.Batch, newHeadMsgIdx arbutil.MessageIndex, newMessages []arbostypes.MessageWithMetadataAndBlockHash) error {
+	lastDelayedMsgIdx, err := s.getPrevPrevDelayedRead(newHeadMsgIdx + 1)
```

Comment:
Kept the same behavior that Nitro has.
But not sure if this should actually be: 
```suggestion
	lastDelayedMsgIdx, err := s.getPrevPrevDelayedRead(newHeadMsgIdx)
```